diffusion/models/stable_diffusion.py

```diff
@@ -193,10 +214,13 @@ def forward(self, batch):
                 text_pooled_embeds *= batch['drop_caption_mask'].view(-1, 1)
 
         # Sample the diffusion timesteps
-        timesteps = torch.randint(0,
-                                  len(self.noise_scheduler), (latents.shape[0],),
-                                  device=latents.device,
-                                  generator=self.rng_generator)
+        if self.quasirandomness:
+            timesteps = self._generate_quasirandom_timesteps(latents)
+        else:
+            timesteps = torch.randint(0,
+                                      len(self.noise_scheduler), (latents.shape[0],),
+                                      device=latents.device,
+                                      generator=self.rng_generator)
```

Comment:
Nit: to me, it would be cleaner to have `_generate_timesteps()` and place this if-statement within that function.